docs/structure.rst

```diff
@@ -0,0 +1,43 @@
+
+Structure of a Reactive Charm
+-----------------------------
+
+The structure of a reactive charm is similar to existing charms, with the
+addition of ``reactive`` directory and the ``relations`` directory under
+``hooks``:
+
+.. code-block:: text
+
+    .
+    ├── metadata.yaml
+    ├── reactive
+    │   └── common.py
+    └── hooks
+        ├── pgsql-relation-changed
+        └── relations
+            └── pgsql
+                ├── interface.yaml
+                ├── peer.py
+                ├── provides.py
+                └── requires.py
+
+The hooks will need to call :func:`reactive.main() <charms.reactive.main>`,
+and the decorated handler blocks can be placed in any file under the ``reactive``
+directory.  Thus, pretty much all of your hooks will end up contain little more
+than:
```

Comment:
It'd be nice to document more of what a reactive charm might contain, such as metrics.yaml, or actions.yaml and actions/